src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/IDE/VSSolutionManager.cs

```diff
@@ -629,73 +632,74 @@ private void SetDefaultProjectName()
 
         private async Task EnsureNuGetAndVsProjectAdapterCacheAsync()
         {
-            ThreadHelper.ThrowIfNotOnUIThread();
-
-            if (!_cacheInitialized && IsSolutionOpen)
+            await _initLock.ExecuteNuGetOperationAsync(async () =>
```

Comment:
But what's the issue with having this lock as well? I still think we should have this for edge cases or worst scenario, it should never start a second initialization.